trie/database.go

```diff
@@ -93,20 +93,18 @@ func prepare(diskdb ethdb.Database, config *Config) *Database {
 
 // NewDatabase initializes the trie database with default settings, namely
 // the legacy hash-based scheme is used by default.
-func NewDatabase(diskdb ethdb.Database) *Database {
-	return NewDatabaseWithConfig(diskdb, nil)
-}
-
-// NewDatabaseWithConfig initializes the trie database with provided configs.
-// The path-based scheme is not activated yet, always initialized with legacy
-// hash-based scheme by default.
-func NewDatabaseWithConfig(diskdb ethdb.Database, config *Config) *Database {
+func NewDatabase(diskdb ethdb.Database, config *Config) *Database {
```

Comment:
How come you did this change? Seems to me the fallout of this is that a lot of callsites now are forced to pass a `nil`, like `NewEmpty(NewDatabase(rawdb.NewMemoryDatabase(), nil))`. 

Just curious why you found it worthwhile to change it...?

Reply:
Originally `NewDatabase` is actually equivalent with `NewDatabase(disk, nil)` as now.
I think it's a bit redundant to have two methods.

But I really need to think about what's the best default config if the pass one is nil.